lib/logflare/backends/adaptor/clickhouse_adaptor/query_templates.ex

```diff
@@ -0,0 +1,208 @@
+defmodule Logflare.Backends.Adaptor.ClickhouseAdaptor.QueryTemplates do
+  @moduledoc """
+  Common query templates utilized by the `ClickhouseAdaptor`.
+  """
+
+  import Logflare.Utils.Guards
+
+  @default_key_type_counts_view_name "mv_key_type_counts_per_minute"
+  @default_key_type_counts_table_name "key_type_counts_per_minute"
+  @default_ttl_days 90
+
+  @doc """
+  Generates a ClickHouse query statement to check that the user GRANTs include the needed permissions.
+
+  The results will return a `1` if the user _has_ the needed GRANTs or a `0` otherwise.
+
+  Because this is generally run via a connection that was provided with the
+  user credentials and database, there is no need to supply the specific DB by default.
+
+  ###Options
+
+  - `:database` - (Optional) Will produce a fully qualified `<database>.*` string when provided with a value. Defaults to `nil`.
+
+  """
+  @spec grant_check_statement(opts :: Keyword.t()) :: String.t()
+  def grant_check_statement(opts \\ []) when is_list(opts) do
+    database = Keyword.get(opts, :database, nil)
+
+    grant_target_string =
+      if is_non_empty_binary(database) do
+        "#{database}.*"
+      else
+        "*"
+      end
+
+    "CHECK GRANT CREATE TABLE, ALTER TABLE, INSERT, SELECT, DROP TABLE, CREATE VIEW, DROP VIEW ON #{grant_target_string}"
+  end
+
+  @doc """
+  Generates a ClickHouse query statement to provision an ingest table for logs.
+
+  ###Options
+
+  - `:database` - (Optional) Will produce a fully qualified `<database>.<table>` string when provided with a value. Defaults to `nil`.
+  - `:ttl_days` - (Optional) Will add a TTL statement to the table creation query. Defaults to `90`. `nil` will disable the TTL.
+
+  """
+  @spec create_log_ingest_table_statement(table :: String.t(), opts :: Keyword.t()) :: String.t()
+  def create_log_ingest_table_statement(table, opts \\ [])
+      when is_non_empty_binary(table) and is_list(opts) do
+    database = Keyword.get(opts, :database)
+    ttl_days_temp = Keyword.get(opts, :ttl_days, @default_ttl_days)
```

Comment:
the ttl should be based off the bigquery retention value on the retention_days value on Source.
should default to 3 days as per free accounts.

Reply:
@Ziinc I think I have this covered now, but please check when you have a moment.